src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -541,6 +558,28 @@ object AppDefinition {
       (cmd ^ args) || (!(cmd || args) && container)
     }
 
+  private val complyWithSchedulerLabelRules: Validator[AppDefinition] =
+    isTrue("Scheduler AppDefinition must define required labels") { app =>
+      val isNormalApp = !app.isScheduler
+      val containsRequiredSchedulerLabels =
+        app.labels.get(AppDefinition.Labels.DcosPackageFrameworkName).exists(_.nonEmpty) &&
+          app.labels.get(AppDefinition.Labels.DcosMigrationApiVersion).contains("v1") &&
```

Comment:
Not sure if we can define this for all schedulers.
Perhaps it makes more sense to say: if the DcosMigrationApiVersion is set, then we expect to see:
-frameworkName
-migrationApiPath

That also means: the whole validator should be called: `complyWithMigrationAPI`